tox.ini

```diff
@@ -12,47 +12,60 @@ skip_missing_interpreters = true
 
 [testenv]
 description = run the tests with pytest under {basepython}
-setenv = COVERAGE_FILE={toxworkdir}/.coverage.{envname}
+setenv = VIRTUALENV_DOWNLOAD = 0
+         PIP_DISABLE_VERSION_CHECK = 1
+         COVERAGE_FILE = {toxworkdir}/.coverage.{envname}
 passenv = http_proxy https_proxy no_proxy SSL_CERT_FILE TOXENV CI TRAVIS TRAVIS_* APPVEYOR APPVEYOR_* CODECOV_*
+deps = setuptools >= 29.0.0
+       setuptools_scm >= 2.0.0, < 3
+       wheel >= 0.31.0
```

Comment:
you set `--no-download,`, but then ask for an upgrade of `wheel` / `setuptools`? (why not just allow virtualenv to do that for you at this point?)

Reply:
It has to do with the index URL used. By default virtualenv uses PyPi. This way the tox config adheres too whatever the user has configured; e.g. value from ``PIP_INDEX_URL``.